### PR TITLE
Updated vagrantfile and readme

### PR DESCRIPTION
**Pull request by johntdyer**
I updated the readme to hopefully help new users, which may not understand the relationship of Vagrant and Chef.  I also removed max_retried and timeout since those appear to be deprecated attributes in Vagrant 1.3.5, and their presence caused vagrant up to fail

``` shell
jdyer@retina:~/Projects/metasploit(master○) » vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
There are errors in the configuration of this machine. Please fix
the following errors and try again:

SSH:
* The following settings shouldn't exist: max_tries, timeout


```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ configuring you're metasploit development environment.
 # Requirements
 ## Platform
 The following systems are supported ***and*** tested on:
-* Mac OS X Mountain Lion and above (10.8+)
+* Mac OS X Mountain Lion and above (10.8+) w/ XCode Developer tools.
```
Comment (erran) [inline README.md]: Shouldn't XCode be Xcode? Developer tool -> Command Line Tools?

File: README.md [hunk 2]
```diff
@@ -13,9 +13,20 @@ already supported. If you're running Windows, feel free to tweak some recipes to
 add support by editing the case-when `node[:platform]` blocks and adding
 `when 'windows'`.
 
+## Vagrant
+* Vagrant 1.2+
+* Vagrant Berkshelf 1.3.2+
```
Comment (erran) [inline README.md:23]: We should also reference bindler which I have added to a few cookbooks already (though they're also **out of date** with the latest Vagrant release.

File: README.md (hunk 1 above)
Comment (johntdyer) [inline README.md]: I might be mistaken but I thought commandline tools was just git, the compiler was in Xcode itself....

Comment (erran) [inline README.md]: I'd say Xcode and Command Line Tools, here's a guide that uses the terminology I'd like to see: http://burnedpixel.com/blog/setting-up-git-and-github-on-your-mac/

File: README.md (hunk 2 above)
Comment (johntdyer) [inline README.md:23]: I don't know bindler.... Do you want me to add that reference ?

**Comment by erran** [comment]
I'll take care of the Bindler updates later. Thanks for this PR.

**Pull request merged**